src/interchange/src/avro/schema.rs

```diff
@@ -448,8 +450,11 @@ impl SchemaCache {
                 // An issue with _fetching_ the schema should be returned
                 // immediately, and not cached, since it might get better on the
                 // next retry.
-                // TODO - some sort of exponential backoff or similar logic
-                let response = self.ccsr_client.get_schema_by_id(id).await?;
+                let ccsr_client = &self.ccsr_client;
+                let response = Retry::default()
```

Comment:
Could there be other cases where calls to  get_schema_by_id will be needed the future? Maybe the retry should be inside get_schema_id?

Reply:
It looks like the only other code that calls this API is test code and we likely don't want to retry in those situations. We could have `get_schema_by_id` take a `Retry` object, or the parameters to a retry object?